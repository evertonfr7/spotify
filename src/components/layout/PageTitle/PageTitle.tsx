import Typography from '@/components/ui/Typography'
import { PageTitleProps } from './PageTitle.types'

export function PageTitle({
  title,
  subtitle,
  children,
}: PageTitleProps): JSX.Element {
  return (
    <div className="flex justify-between items-center flex-wrap mb-4 md:mb-8">
      <div className="flex flex-col gap-2">
        <Typography variant="title" as="h1">
          {title}
        </Typography>
        {subtitle && (
          <Typography className="mb-2 md:mb-0" variant="subtitle" as="h2">
            {subtitle}
          </Typography>
        )}
      </div>
      {children && <div>{children}</div>}
    </div>
  )
}
