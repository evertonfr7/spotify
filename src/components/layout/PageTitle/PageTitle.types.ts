export interface PageTitleProps {
  title: string
  subtitle?: string
  children?: React.ReactNode
}
