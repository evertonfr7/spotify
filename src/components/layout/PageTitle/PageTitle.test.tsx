import { render, screen } from '@testing-library/react'
import { PageTitle } from './PageTitle'

describe('PageTitle Component', () => {
  test('renders the page title with correct snapshot', () => {
    const tree = render(
      <PageTitle title="Test Title" subtitle="Test Subtitle" />,
    )

    expect(tree).toMatchSnapshot()
  })
  test('renders title correctly', () => {
    render(<PageTitle title="Test Title" />)
    expect(screen.getByText('Test Title')).toBeInTheDocument()
  })

  test('renders subtitle when provided', () => {
    render(<PageTitle title="Test Title" subtitle="Test Subtitle" />)
    expect(screen.getByText('Test Subtitle')).toBeInTheDocument()
  })

  test('does not render subtitle when not provided', () => {
    render(<PageTitle title="Test Title" />)
    expect(screen.queryByText('Test Subtitle')).toBeNull()
  })

  test('renders children when provided', () => {
    render(
      <PageTitle title="Test Title">
        <button>Click Me</button>
      </PageTitle>,
    )
    expect(screen.getByRole('button', { name: 'Click Me' })).toBeInTheDocument()
  })

  test('does not render children when not provided', () => {
    render(<PageTitle title="Test Title" />)
    expect(screen.queryByRole('button')).toBeNull()
  })

  test('renders with correct structure', () => {
    render(<PageTitle title="Test Title" subtitle="Test Subtitle" />)
    const title = screen.getByText('Test Title')
    const subtitle = screen.getByText('Test Subtitle')

    expect(title).toBeInTheDocument()
    expect(subtitle).toBeInTheDocument()
    expect(title.tagName.toLowerCase()).toBe('h1') // Presume que Typography com `variant="title"` usa `h1`.
    expect(subtitle.tagName.toLowerCase()).toBe('h2') // Presume que Typography com `variant="subtitle"` usa `h2`.
  })
})
