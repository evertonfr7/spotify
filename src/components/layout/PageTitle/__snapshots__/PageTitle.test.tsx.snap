// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle Component renders the page title with correct snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-between items-center flex-wrap mb-4 md:mb-8"
      >
        <div
          class="flex flex-col gap-2"
        >
          <h1
            class="font-default text-[26px]/[30px] font-semibold md:text-[28px]/[32px]"
          >
            Test Title
          </h1>
          <h2
            class="font-default text-[14px]/[16px] font-normal text-[#D3DADD] md:text-base/[18px] mb-2 md:mb-0"
          >
            Test Subtitle
          </h2>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-between items-center flex-wrap mb-4 md:mb-8"
    >
      <div
        class="flex flex-col gap-2"
      >
        <h1
          class="font-default text-[26px]/[30px] font-semibold md:text-[28px]/[32px]"
        >
          Test Title
        </h1>
        <h2
          class="font-default text-[14px]/[16px] font-normal text-[#D3DADD] md:text-base/[18px] mb-2 md:mb-0"
        >
          Test Subtitle
        </h2>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
